posthog/async_migrations/definition.py

```diff
@@ -24,50 +25,43 @@ class AsyncMigrationType:
 
 class AsyncMigrationOperation:
     def __init__(
-        self,
-        fn: Callable[[str], None],
-        rollback_fn: Callable[[str], None] = lambda _: None,
-        debug_context: Optional[Any] = None,
+        self, fn: Callable[[str], None], rollback_fn: Callable[[str], None] = lambda _: None,
     ):
         self.fn = fn
 
         # This should not be a long operation as it will be executed synchronously!
         # Defaults to a no-op ("") - None causes a failure to rollback
         self.rollback_fn = rollback_fn
 
-        self.debug_context = debug_context
 
-    @classmethod
-    def simple_op(
-        cls,
-        sql,
-        rollback=None,
-        database: AnalyticsDBMS = AnalyticsDBMS.CLICKHOUSE,
-        timeout_seconds: int = ASYNC_MIGRATIONS_DEFAULT_TIMEOUT_SECONDS,
-    ):
-        return cls(
-            fn=cls.get_db_op(database=database, sql=sql, timeout_seconds=timeout_seconds),
-            rollback_fn=cls.get_db_op(database=database, sql=rollback) if rollback else lambda _: None,
-            debug_context={"sql": sql, "rollback": rollback, "database": database},
-        )
-
-    @classmethod
-    def get_db_op(
-        cls,
+class AsyncMigrationOperationSQL(AsyncMigrationOperation):
```

Comment:
👍 for this refactor

Reply:
Tests are failing though: the issue is that the behavior of "what should happen if rollback sql is None" was previously unspecified. I think it was automatically supposed to "pass", but can you confirm this?

Reply:
hmm in the past this was discussed and the idea was that unspecified rollbacks should indeed error so that people catch this issue during development, as we'd like to have rollbacks for everything.

The solution would be to explicitly set `rollback=""`. 

We can certainly discuss changing this behavior though

Reply:
The APIs here are kind of badly designed in that we're assuming users _need_ to pass in arguments but don't enforce that language-level. There's easy ways to fix that but this will balloon the scope of this PR - splitting them out into a separate PR, will post soon.